src/software/jetson_nano/thunderloop_main.cpp

```diff
@@ -18,16 +29,78 @@ std::string BANNER =
 "  /'                                                                                       /'          \n";
 // clang-format on
 
+
+/*
+ * Configure malloc for real-time linux
+ *
+ * https://rt.wiki.kernel.org/index.php/Dynamic_memory_allocation_example
+ */
+static void configureMallocBehaviour(void)
+{
+    // Now lock all current and future pages
+    // from preventing of being paged
+    if (mlockall(MCL_CURRENT | MCL_FUTURE))
+    {
+        perror("mlockall failed:");
+    }
+
+    // Turn off malloc trimming.
+    mallopt(M_TRIM_THRESHOLD, -1);
+
+    // Turn off mmap usage.
+    mallopt(M_MMAP_MAX, 0);
+}
+
+
+/*
+ * Reserve memory for this process
+ *
+ * @param size How many bytes to reserve
+ */
+static void reserveProcessMemory(int size)
+{
+    long int i;
+    char* buffer;
+
+    buffer = static_cast<char*>(malloc(size));
+
+    // Touch each page in this piece of memory to get it mapped into RAM
+    for (i = 0; i < size; i += sysconf(_SC_PAGESIZE))
+    {
+        // Each write to this buffer will generate a pagefault.
+        // Once the pagefault is handled a page will be locked in
+        // memory and never given back to the system.
+        buffer[i] = 0;
+    }
+
+    // Buffer will now be released. As Glibc is configured such that it
+    // never gives back memory to the kernel, the memory allocated above is
+    // locked for this process. All malloc() and new() calls come from
+    // the memory pool reserved and locked above. Issuing free() and
+    // delete() does NOT make this locking undone. So, with this locking
+    // mechanism we can build C++ applications that will never run into
+    // a major/minor pagefault, even with swapping enabled.
+    free(buffer);
+}
+
 int main(int argc, char** argv)
 {
     std::cout << BANNER << std::endl;
 
+    // Page faults are bad, lets setup malloc and reserve some memory
+    configureMallocBehaviour();
+
+    // 100MB pagefault free buffer
+    const int pre_allocation_size = 20 * 1024 * 1024;
```

Comment:
any way to check if we exceed this?

Reply:
not that I am aware of, its not catastrophic if it goes over, we won't be realtime and the thunderloop status should indicate that